docs/reference/quickstart/getting-started.asciidoc

```diff
@@ -193,31 +222,100 @@ You should receive a response indicating there were no errors.
 ===============
 
 [discrete]
-[[qs-search-data]]
+[[getting-started-mappings-and-data-types]]
+=== Define mappings and data types
+
+<<elasticsearch-intro-documents-fields-mappings,Mappings>> define how data is stored and indexed in {es}, much like a  schema for a relational database.
+
+[discrete]
+[[getting-started-dynamic-mapping]]
+==== Use dynamic mapping
+
+When using dynamic mapping, {es} automatically creates mappings for new fields by default.
+The documents we've added so far have used dynamic mapping, because we didn't specify a mapping when creating the index.
+
+Let's add a new document to the `books` index with a field that doesn't exist, to see how dynamic mapping works.
+
+[source,console]
+----
+POST /books/_doc
+{
+  "name": "The Great Gatsby",
+  "author": "F. Scott Fitzgerald",
+  "release_date": "1925-04-10",
+  "page_count": 180,
+  "new_field": "This is a dynamically added field"
+}
+----
+// TEST[continued]
+
+View the mapping for the `books` index. The new field `new_field` has been added to the mapping with a `text` data type.
+
+[source,console]
+----
+GET /books/_mapping
+----
+// TEST[continued]
+
+[discrete]
+[[getting-started-explicit-mapping]]
+==== Define explicit mapping
+
+Create an index named `my-explicit-mappings-books` with explicit mappings:
+
+[source,console]
+----
+PUT /my-explicit-mappings-books
+{
+  "mappings": {
+    "dynamic": false,  <1>
+    "properties": {
+      "name": { "type": "text" },
+      "author": { "type": "text" },
+      "release_date": { "type": "date", "format": "yyyy-MM-dd" },
+      "page_count": { "type": "integer" }
+    }
+  }
+}
+----
+// TEST[continued]
+<1> Disables dynamic mapping for the index. Documents containing fields not defined in the mapping will be rejected.
+
+[discrete]
+[[getting-started-combined-mapping]]
+==== Combine dynamic and explicit mappings
+
+When an index has the `dynamic` flag set to `true`, you can add new fields to documents without updating the mapping.
+This allows you to combine dynamic and explicit mappings.
+
+[discrete]
+[[getting-started-search-data]]
 === Search data
 
 Indexed documents are available for search in near real-time.
+// TODO: You'll find more detailed quick start guides in TODO
 
 [discrete]
-[[search-all-documents]]
+[[getting-started-search-all-documents]]
```

Comment:
again don't want to litter everything with responses but should the search data examples show what to expect since they are get commands?

Reply:
great question and I'm not sure because we would have to collapse them to make the page readable, so will need to find consensus after feedback

it might be easier to not print responses and put the onus on the user to actually run the examples which is the whole point...but it is a balancing act

Reply:
I would like collapsible results here, or truncated ones

Reply:
Due to high demand, I've gone all-in on responses and annotations